pkg/apis/vpc/v1alpha1/subnetipreservation_types.go

```diff
@@ -0,0 +1,64 @@
+/* Copyright © 2025 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetIPReservationSpec defines the desired state of SubnetIPReservation
+type SubnetIPReservationSpec struct {
+	// Subnet defines the parent Subnet name of the SubnetIPReservation.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="Subnet is immutable"
+	Subnet string `json:"subnet"`
+
+	// RequestedNumberOfIPs defines number of IPs requested to be reserved.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="RequestedNumberOfIPs is immutable"
+	// +kubebuilder:validation:Maximum:=100
+	// +kubebuilder:validation:Minimum:=1
+	RequestedNumberOfIPs int `json:"requestedNumberOfIPs"`
```

Comment:
This field is not optional?

Reply:
What is the NSX API behavior?

Reply:
Sure, let me confirm with the NSX side for the behavior

Reply:
Ok.